### PR TITLE
add 'sequence' benchmark

### PR DESCRIPTION
**Pull request by c-cube**
first attempt at adding basic micro benchmarks from my `sequence` library

### CODE REVIEW

**Comment by chambart** [comment]
Thank you.

File: benchmarks/sequence/benchmark.ml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+let id x = x
+
+(* no validation *)
+let valid_trivial _n _res = M.Ok
+
+let bench_map_fold =
+  "map_fold", M.Int (map_fold, id, valid_trivial,
+                     [M.Range (100, 100), M.Short
+                     ;M.Range (100_000, 100_000), M.Short])
```
Comment (chambart) [inline benchmarks/sequence/benchmark.ml]: Why did you put fixed ranges here ?

**Comment by c-cube** [comment]
Note that I'm not sure exactly how to use operf-micro. I was not sure of the semantics of range: if I write `M.Range (1, 100)`, will it try every integer in the range, or a random one?

File: benchmarks/sequence/benchmark.ml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
```
Comment (chambart) [inline benchmarks/sequence/benchmark.ml]: It is also probably a good idea to have a 'baseline' comparison that tells what is the best possible equivalent code, see for instance: https://github.com/OCamlPro/operf-micro/blob/master/benchmarks/lens/benchmark.ml#L62

**Comment by chambart** [comment]
Thanks !

**Pull request merged**